os-eap-txnrecovery/added/partitionPV.sh

```diff
@@ -71,38 +71,41 @@ function startApplicationServer() {
   $IS_TX_SQL_BACKEND && initJdbcRecoveryMarkerProperties
   $IS_SPLIT_DATA_DEFINED && mkdir -p "${podsDir}"
 
-  # 2) while any recovery marker matches, sleep and wait for recovery to finish
-  local waitCounter=0
-  while true; do
-
-    if isRecoveryInProgress; then
-      log_info "[`date`] Waiting to start pod ${POD_NAME} as recovery process for the pod is currently in progress"
-    else
-      # no recovery running: we are free to start the app container
-      break
-    fi
-    sleep 1
-  done
+  # allow this to be skipped if we're testing, or running in docker with no db etc.
+  if [ "x${JDBC_SKIP_RECOVERY}" != "xtrue" ]; then
```

Comment:
@ochaloup This is an additional change you should be aware of. The testsuite at the moment won't be able to get past this point of pgsql env vars are present, as the recovery process will block. This env var allows this to be skipped.

(In the case of the behave test suite, the container is booted and the configuration is checked for certain items present, there is never an actual postgres instance running. In the ARQ test suite there is.)

We can change this to have a different name if you like, but liek the other change, the only user is the testsuite.

Reply:
ok, this seems fine to me. Just thinking if it could be done in a nicer way. But I don't have particular idea currently. Especially that I haven't got to run the testsuite yet, unfortunately.

Reply:
@ochaloup Ping me on IRC if you want some help with that, the behave testsuite should run just about anywhere.